classes/mix.bbclass

```diff
@@ -0,0 +1,182 @@
+            hex_src_uri = ";".join((hex_uri,
+                'subdir=hex-deps/%s'%key,
+                'name=%s'%key))
+            d.appendVar('SRC_URI', " %s "%hex_src_uri)
+            md5, sha256 = checksum(hex_uri)
+            d.setVarFlag("SRC_URI", "%s.md5sum"%key, md5)
+            d.setVarFlag("SRC_URI", "%s.sha256sum"%key, sha256)
+            bb.debug(2, 'SRC_URI %s '%hex_src_uri)
+            bb.debug(2, "SRC_URI %s.md5sum %s"%(key, md5))
+            bb.debug(2, "SRC_URI %s.sha256sum %s"%(key, sha256))
+
+do_compile() {
+    cp -avl ${WORKDIR}/git-deps ./deps
+    for mixdep in ${WORKDIR}/hex-deps/*; do
+        mixdepbase="`basename $mixdep`"
+        mkdir -pv deps/$mixdepbase
+        tar xz -C deps/$mixdepbase -f $mixdep/contents.tar.gz
+    done
+
+    export LC_ALL=en_US.UTF-8
+    export MIX_ENV=release
+
+    # explicitly use TARGET erts instead of the erlang-native provided one:
+    TARGET_ERTS="${STAGING_DIR_TARGET}/usr/lib/erlang"
+    mkdir -pv rel
+
+    echo "{include_erts, \"${TARGET_ERTS}\"}." > rel/relx.config
+
+    echo "{lib_dirs, [" >> rel/relx.config
+        echo "\"${STAGING_DIR_TARGET}/usr/lib/erlang/lib/lager_journald_backend-*\"," >> rel/relx.config
+        echo "\"${STAGING_DIR_TARGET}/usr/lib/erlang/lib/ejournald-*\"" >> rel/relx.config
+    echo "]}." >> rel/relx.config
+
+    echo "generated relx.config:"
+    echo "==============="
+    cat rel/relx.config
+    echo "==============="
+
+    mix do deps.compile, compile, release
+}
+
+do_install() {
+    if [ -e rel/${REL_NAME}/releases/${REL_VSN}/${REL_NAME}.tar.gz ]
+    then
+        TAR_DIR="rel/${REL_NAME}/releases/${REL_VSN}/${REL_NAME}.tar.gz"
+        bbnote "tar found at rel/${REL_NAME}/releases/${REL_VSN}/${REL_NAME}.tar.gz"
+    elif [ -e rel/${REL_NAME}/${REL_NAME}-${REL_VSN}.tar.gz ]
+    then
+        TAR_DIR="rel/${REL_NAME}/${REL_NAME}-${REL_VSN}.tar.gz"
+        bbnote "tar found at rel/${REL_NAME}-${REL_VSN}.tar.gz"
+    else
+        bbfatal "${REL_NAME}: tar file not found"
+        exit 1
+    fi
+
+    install -m 0755 -d "${D}/${APP_PREFIX}/${APPNAME}/${APPVERSION}/"
+    tar xvz -C ${D}/${APP_PREFIX}/${APPNAME}/${APPVERSION} -f $TAR_DIR
+    ln -s ${REL_NAME} "${D}/${APP_PREFIX}/${APPNAME}/${APPVERSION}/bin/rc"
+
+    install -m 0755 -d "${D}/${SYSCONFIG_PREFIX}"
+    install -m 0644 ${S}/config/${APPNAME}.conf ${D}/${SYSCONFIG_PREFIX}/${APPNAME}.conf
+    echo "log.journal.level = info" >> ${D}/${SYSCONFIG_PREFIX}/${APPNAME}.conf
+
+    install -d ${D}${systemd_unitdir}/system/
+    install -m 0644 ${MIX_UNIT_FILE}/mix.service ${D}${systemd_unitdir}/system/${SYSTEMD_UNIT_NAME}.service
+    sed -i "s/@@DESCRIPTION@@/${DESCRIPTION}/" ${D}${systemd_unitdir}/system/${SYSTEMD_UNIT_NAME}.service
+    sed -i "s/@@VERSION@@/${APPVERSION}/" ${D}${systemd_unitdir}/system/${SYSTEMD_UNIT_NAME}.service
+}
+
+python do_mix_deps() {
+        gen_deps(d)
+        src_uri = (d.getVar('SRC_URI', True) or "").split()
```

Comment:
base_do_fetch() does the same thing, either reuse it or at least replicate the error checks